src/proof/arith_proof_recorder.h

```diff
@@ -0,0 +1,101 @@
+/*********************                                                        */
+/*! \file arith_proof_recorder.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Alex Ozdemir
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief A class for saving the skeletons of a arithmetic proofs for later.
+ **
+ ** In particular, we're interested in proving bottom from a conjunction of
+ ** theory literals.
+ **
+ ** For now, we assume that this can be done using a Farkas combination, and if
+ ** that doesn't work for some reason, then we give up and "trust" the lemma.
+ ** In the future we'll build support for more sophisticated reasoning.
+ **
+ ** Given this scope, our task is to...
+ **   for each lemma (a set of literals)
+ **   save the farkas coefficients for those literals
+ **      which requires we save an ordering of the literals
+ **      and a parallel ordering of farkas coefficients.
+ **
+ ** Farkas proofs have the following cocre structure:
+ **    For a list of affine bounds: c[i] dot x >= b[i]
+ **    and a list of non-negative coefficients: f[i],
+ **    compute
+ **
+ **             sum_i{ (c[i] dot x) * f[i] }     and   sum_i{b[i]*f[i]}
+ **
+ **    and then verify that the left is actually < the right, a (=><=)
```

Comment:
1. Is `x` known at this stage?
2. What is `(=><=)`?
3. Where is this check done in the code?

Reply:
1. I'll add a definition -- it's a vector of variables.
2. It represents contradiction. I'll replace it with the word.
3. This code doesn't check proofs, it just stores them. I'll add that to the documentation.